site/eslint.config.js

```diff
@@ -106,18 +106,24 @@ module.exports = tseslint.config(
     {
         name: 'Options for cypress files',
         files: ['cypress/**/*.{js,ts}'],
-        extends: [cypress.configs.globals],
-        // todo: enable cypress lint rules at some point
-        // extends: [cypress.configs.recommended],
+        extends: [cypress.configs.recommended],
         languageOptions: {
             globals: globals.nodeBuiltin,
         },
+        rules: {
+            'no-restricted-syntax': [
+                'error',
+                {
+                    selector: '[type=CallExpression][callee.object.name=cy][callee.property.name=waitAndReloadUntil]',
+                    message: 'Do not wait for arbitrary time periods',
+                },
+            ],
+        },
```

Comment:
While this might be a desirable change eventually, I don't think it's necessarily related to the basic task of linting the tests.  As @IDzyre noted, we can't just blindly remove waits in some cases.  While some of the waits may be unnecessary, many others were specifically added to address flakiness issues and I worry that removing them may cause undesirable consequences.

Reply:
This rule serves the same purpose cypress's built-in [no-unnecessary-waiting](https://github.com/cypress-io/eslint-plugin-cypress/blob/master/docs/rules/no-unnecessary-waiting.md) rule, with the only difference being that it applies to our custom `cy.waitAndReloadUntil()` function rather than cypress's `cy.wait()` function. Since `cy.waitAndReloadUntil()` uses `cy.wait()`, I believe usage of `cy.waitAndReloadUntil()` should be beholden to the same standard as `cy.wait()`. The point of these rules is to discourage the usage of these functions; if a test needs them for any reason, a comment can be put explaining why, along with an `eslint-disable-next-line` comment to disable the relevant rule as a one time exception.

For examples of each, look at the following:
https://github.com/Submitty/Submitty/pull/10562/files#diff-3d193f8781cad654997675b84fc39f55904d8c13d9e929433a8ac82aabb3c2e3R84-R87
https://github.com/Submitty/Submitty/pull/10562/files#diff-caa8c69680af44607887452b399ee8414370cd89028b959f3c3036c3bebf4998R54-R58

Additionally, it seems that all of the `wait`s this PR does remove were actually not needed, since the tests they were removed from repeatedly pass CI checks (this can be seen in my commit history, there were only two tests failing that I had to fix).